rpc/src/v1/impls/eth.rs

```diff
@@ -439,6 +440,10 @@ impl<C, SN: ?Sized, S: ?Sized, M, EM, T: StateInfo + 'static> EthClient<C, SN, S
 		Ok(Some(block))
 	}
 
+	/// Get state for the given block number. Returns either the State or a block from which state
+	/// can be retrieved.
+	/// Note: When passing `BlockNumber::Pending` we fall back to the state of the current best block
```

Comment:
I'm actually not sure if fallback behavior is desired. I'd rather return a more meaningful error which underlines that:
1. Pending state is only available if mining or using `--force-sealing`.
2. Pending state is discouraged cause it's not available for light clients.

If a dapp requests `pending` state and the client doesn't really have one it seems more like some errornous setup for me. In the past we did fall back to `latest` afair (at least for some methods) - maybe there is still a CLI flag that controls this behavior?

Reply:
> it seems more like some errornous setup for me

Me too. However this is the way MyCrypto works (and many other users too, given the amount of bug reports we get on this)  and they have been unable to fix their code for quite a long while. The behaviour of `get_state()` in this regard is unchanged in this PR and does indeed fall back to `"latest"`. This PR adds this fallback behaviour to `eth_call()` and `eth_estimateGas` as well.

Reply:
Fair enough!